core-dev/packages/coq/coq.8.12.dev/opam

```diff
@@ -33,7 +33,8 @@ build: [
     "-libdir" "%{lib}%/coq"
     "-datadir" "%{share}%/coq"
     "-coqide" "no"
-    "-flambda-opts" "-O3 -unbox-closures"
+    # These settings are used as templates for release packages
```

Comment:
I don't understand the point of this comment here. If native-compiler is broken on macOS, I'd expect a comment pointing to why it is broken. Whether or not they are sued for release packages is irrelevant IMO; even dev packages should work on macos after all.

Reply:
Likewise, "unbox-closures" is not set because it is broken, so the justification should be "here's the bug", not "this is a template for releases".

Reply:
True. My comments was meant for the current users of these packages, whose revealed preferences seem different.

This PR could also wait for https://github.com/ocaml/opam-repository/pull/16887 to be finalized and sync with that, since the solution there should be much better, and would be needed even if we fixed all current bugs.

Reply:
That seems to still enable flambda for ocaml >= 4.07 -- so only older ocaml is broken?

Indeed if that PR is likely to be accepted, it makes sense to wait for it.

Reply:
That enables flambda *if you picked an flambda compiler* (else those options are ignored).

> so only older ocaml is broken?

flambda+native is also "YMMV", depending on multiple factors, but native must also be selected explicitly.

The story for native is even longer and I'll spare you, but the plan (in https://github.com/coq/coq/issues/12757) includes producing sufficient docs; eventually, enabling native should precompile native versions of all libraries where it makes sense.